### ISSUE TITLE
"rm" fails on group's that have children or parents

### ISSUE THREAD

**Issue by RusDavies** [RusDavies/moose-inventory#11]
For example, the following throws an error:

``` shell
 moose-inventory group add test
 moose-inventory group addchild test testchild
 moose-inventory.sh group rm testchild
```

The output of the last command, with relevant error, is as follows: 

``` shell
Remove group 'testchild':
  - Retrieve group 'testchild'...
    - OK
  - Destroy group 'testchild'...
/home/russell/.gem/ruby/gems/sqlite3-1.3.10/lib/sqlite3/database.rb:230:in `step': SQLite3::ConstraintException: FOREIGN KEY constraint failed (Sequel::ForeignKeyConstraintViolation)
    from /home/russell/.gem/ruby/gems/sqlite3-1.3.10/lib/sqlite3/database.rb:230:in `block in execute_batch'
⋮ 
```

**Comment by RusDavies** [comment]
By contrast, the following, using `rmchild` removes the association, but leaves the child in place & orphaned, which is the intended behaviour.  Point being, we are able to remove associations. 

``` shell
 moose-inventory group add test
 moose-inventory group addchild test testchild
 moose-inventory.sh group rmchild test testchild
```

Subsequently, the following then works as expected, since testchild no longer has the blocking associations: 

``` shell
moose-inventory group rm testchild
```

The group 'rm' command should remove the target's associations, prior to removing the target.

**Comment by RusDavies** [comment]
This raises the question of what the behaviour should be when deleting a group with children.  Should child groups simply be dissociated?  Should we propagate the deletion to orphaned children? 

Hosts are allowed to be orphans, being automatically associated with the "ungrouped" group.

Root groups are, by definition, orphans. Orphaning is also the current behaviour when using `rmgroup`. However, is that really what we want?  Should both `rmgroup` and `rm` delete orphaned child groups?  The use case would be the expedient deletion of an entire group hierarchy.  

Perhaps there should be a switch for this?

**Comment by RusDavies** [comment]
The issue is fixed in release 1.0.8.  However, the question of recursively deleting orphaned children has not been addressed.  Instead, that should be the subject of a separate feature request.
